contrib/face_recognition.lua

```diff
@@ -54,39 +64,101 @@ local function _(msgid)
-                        "~/.config/darktable/face_recognition")                                 -- default                                                                  -- default
-                        
-local function show_status (storage, image, format, filename, number, total, high_quality, extra_data)
-  dt.print("Export to Face recognition "..tostring(number).."/"..tostring(total))
+-- preferences
+
+if not dt.preferences.read(MODULE, "initialized", "bool") then
+  dt.preferences.write(MODULE, "unknown_tag", "string", "unknown_person")
+  dt.preferences.write(MODULE, "ignore_tags", "string", "")
+  dt.preferences.write(MODULE, "tolerance", "float", 0.6)
+  dt.preferences.write(MODULE, "num_cores", "integer", 0)
+  dt.preferences.write(MODULE, "known_image_path", "directory", dt.configuration.config_dir .. "/face_recognition")
+  dt.preferences.write(MODULE, "export_format", "integer", 1)
+  dt.preferences.write(MODULE, "max_width", "integer", 1000)
+  dt.preferences.write(MODULE, "max_height", "integer", 1000)
+  dt.preferences.write(MODULE, "initialized", "bool", true)
+end
+
+local function build_image_table(images)
+  local image_table = {}
+  local file_extension = ""
+  local tmp_dir = dt.configuration.tmp_dir .. "/"
+  local ff = fc.export_format.value
+  local cnt = 0
+
+  -- check for plugin-data and direct_edit and build image table accordingly
+
+  if string.match(ff, "JPEG") then
+    file_extension = ".jpg"
+  elseif string.match(ff, "PNG") then
+    file_extension = ".png"
+  elseif string.match(ff, "TIFF") then
+    file_extension = ".tif"
+  end
+
+  for _,img in ipairs(images) do
+    image_table[img] = tmp_dir .. df.get_basename(img.filename) .. file_extension
+    cnt = cnt + 1
+  end
+
+  return image_table, cnt
+end
+
+local function stop_job(job)
+  job.valid = false
+end
+
+local function do_export(img_tbl)
+  local exporter = nil
+  local upsize = false
+  local upscale = false
+  local ff = fc.export_format.value
+  local height = dt.preferences.read(MODULE, "max_height", "integer")
+  local width = dt.preferences.read(MODULE, "max_width", "integer")
+  local images = 0
+  for k,v in pairs(img_tbl) do
+    images = images + 1
+  end
+
+  -- get the export format parameters
+  if string.match(ff, "JPEG") then
+    exporter = dt.new_format("jpeg")
+    exporter.quality = 80
+  elseif string.match(ff, "PNG") then
+    exporter = dt.new_format("png")
+    exporter.bpp = 8
+  elseif string.match(ff, "TIFF") then
+    exporter = dt.new_format("tiff")
+    exporter.bpp = 8
+  end
+  exporter.max_height = height
+  exporter.max_width = width
+
+  -- export the images
+  local job = dt.gui.create_job("export images", true, stop_job)
+  local exp_cnt = 0
+  local percent_step = 1.0 / images
+  job.percent = 0.0
+  for img,export in pairs(img_tbl) do
+    exp_cnt = exp_cnt + 1
+    dt.print(_("Exporting image") .. exp_cnt .. _(" of ") .. images)
```

Comment:
We should use string.format(_("Exporting image %i of %i images."), exp_cnt, images) here

Reply:
Yes, that's a much cleaner way to write that...